include/openthread/tcp.h

```diff
@@ -0,0 +1,616 @@
+ * "end of stream" condition, and future writes on this connection endpoint
+ * will fail.
+ *
+ * The "end of stream" condition only applies after any data previously
+ * provided to the TCP stack to send out has been received by the connection
+ * peer.
+ *
+ * @param[in]  aInstance  A pointer to the OpenThread instance to which the TCP endpoint belongs.
+ * @param[in]  aEndpoint  A pointer to the TCP endpoint structure representing the TCP endpoint to shut down.
+ *
+ * @retval OT_ERROR_NONE    Successfully queued the "end of stream" condition for transmission.
+ * @retval OT_ERROR_FAILED  Failed to queue the "end of stream" condition for transmission.
+ */
+otError otTcpSendEndOfStream(otInstance *aInstance, otTcpEndpoint *aEndpoint);
+
+/**
+ * Forcibly ends the TCP connection associated with this TCP endpoint. This
+ * immediately makes the TCP endpoint free for use for another connection and
+ * empties the send and receive buffers, transferring ownership of any data
+ * provided by the application in @p otTcpSendByReference calls back to
+ * the application. The TCP endpoint's callbacks and memory for the receive
+ * buffer remain associated with the TCP endpoint.
+ *
+ * @param[in]  aInstance  A pointer to the OpenThread instance to which the TCP endpoint belongs.
+ * @param[in]  aEndpoint  A pointer to the TCP endpoint structure representing the TCP endpoint to abort.
+ *
+ * @retval OT_ERROR_NONE    Successfully aborted the TCP endpoint's connection.
+ * @retval OT_ERROR_FAILED  Failed to abort the TCP endpoint's connection.
+ */
+otError otTcpAbort(otInstance *aInstance, otTcpEndpoint *aEndpoint);
+
+/**
+ * Deinitializes this TCP endpoint. This means that OpenThread no longer keeps
+ * track of this TCP endpoint and deallocates all resources it has internally
+ * allocated for this TCP endpoint. The application can reuse the memory
+ * backing the TCP endpoint as it sees fit.
+ *
+ * If it corresponds to a live TCP connection, the connection is terminated
+ * unceremoniously (as in @p otTcpAbort). All resources the application has
+ * provided for this TCP endpoint (linked buffers for the send buffer, memory
+ * for the receive buffer, the @p Endpoint structure itself, etc.) are
+ * immediately returned to the application.
+ *
+ * @param[in]  aInstance  A pointer to the OpenThread instance to which the TCP endpoint belongs.
+ * @param[in]  aEndpoint  A pointer to the TCP endpoint structure to deinitialize.
+ *
+ * @retval OT_ERROR_NONE    Successfully deinitialized the TCP endpoint.
+ * @retval OT_ERROR_FAILED  Failed to deinitialize the TCP endpoint.
+ */
+otError otTcpEndpointDeinitialize(otInstance *aInstance, otTcpEndpoint *aEndpoint);
+
+struct otTcpListener;
+typedef struct otTcpListener otTcpListener;
+
+#define OT_TCP_DEFER  ((otTcpEndpoint *) NULL)
+#define OT_TCP_REFUSE ((otTcpEndpoint *) -1)
+
+/**
+ * This callback indicates that an incoming connection that matches this
```

Comment:
nit: have brief in separate block above

Reply:
Done. I'm not 100% what you meant by "in separate block", can you please check that what I've done is acceptable?